packages/uauth/src/index.ts

```diff
@@ -0,0 +1,384 @@
+                  if (
+                    !chainIdObj.hasOwnProperty('chainId') ||
+                    typeof chainIdObj['chainId'] === 'undefined'
+                  ) {
+                    throw new ProviderRpcError({
+                      code: ProviderRpcErrorCode.INVALID_PARAMS,
+                      message: `The Provider requires a chainId to be passed in as an argument`
+                    })
+                  }
+                  return this.connector.updateSession({
+                    chainId: chainIdObj.chainId,
+                    accounts: this.connector.accounts
+                  })
+                }
+
+                // @ts-ignore
+                if (method === 'eth_sendTransaction') {
+                  // @ts-ignore
+                  return this.connector.sendTransaction(params[0])
+                }
+
+                // @ts-ignore
+                if (method === 'eth_signTransaction') {
+                  // @ts-ignore
+                  return this.connector.signTransaction(params[0])
+                }
+
+                // @ts-ignore
+                if (method === 'personal_sign') {
+                  // @ts-ignore
+                  return this.connector.signPersonalMessage(params)
+                }
+
+                // @ts-ignore
+                if (method === 'eth_sign') {
+                  // @ts-ignore
+                  return this.connector.signMessage(params)
+                }
+
+                // @ts-ignore
+                if (method.includes('eth_signTypedData')) {
+                  // @ts-ignore
+                  return this.connector.signTypedData(params)
+                }
+
+                if (method === 'eth_accounts') {
+                  return this.connector.sendCustomRequest({
+                    id: 1337,
+                    jsonrpc: '2.0',
+                    method,
+                    params
+                  })
+                }
+
+                const chainId = await this.request({ method: 'eth_chainId' })
+
+                if (!this.providers[chainId]) {
+                  const currentChain = chains.find(({ id }) => id === chainId)
+
+                  if (!currentChain) {
+                    throw new ProviderRpcError({
+                      code: ProviderRpcErrorCode.CHAIN_NOT_ADDED,
+                      message: `The Provider does not have a rpcUrl to make a request for the requested method: ${method}`
+                    })
+                  }
+
+                  this.providers[chainId] = new StaticJsonRpcProvider(
+                    currentChain.rpcUrl
+                  )
+                }
+
+                return this.providers[chainId].send(
+                  method,
+                  // @ts-ignore
+                  params
+                )
+              }
+            }
+          }
+          provider = new EthProvider({ chains, connector })
+        }
+        return { provider: provider }
```

Comment:
```suggestion
        return { 
          provider,
          instance: { 
              // SDK methods for the dapp dev to work with user metaData
          }
        }
```

Reply:
There isn’t much a dapp dev can do to manipulate the actual user data (I.e. they cant change the data on their end as we don’t do delegated management), they can really just read the user data and use it throughout the UI in any way they see fit. With that said, I'm just using the instance to pass through the object that was made during login. Definitely simplified a lot of code!